src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -0,0 +1,313 @@
+/*
+* Copyright (c) 2023, NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.nvidia.spark.rapids.jni;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.zone.ZoneOffsetTransition;
+import java.time.zone.ZoneRules;
+import java.time.zone.ZoneRulesException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.concurrent.*;
+
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.DType;
+import ai.rapids.cudf.HostColumnVector;
+import ai.rapids.cudf.Table;
+
+public class GpuTimeZoneDB {
+
+  public static final int TIMEOUT_SECS = 300;
+
+
+  // For the timezone database, we store the transitions in a ColumnVector that is a list of 
+  // structs. The type of this column vector is:
+  //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32>>
+  private CompletableFuture<Map<String, Integer>> zoneIdToTableFuture;
+  private CompletableFuture<HostColumnVector> fixedTransitionsFuture;
+
+  private boolean closed = false;
+
+  GpuTimeZoneDB() {
+    zoneIdToTableFuture = new CompletableFuture<>();
+    fixedTransitionsFuture = new CompletableFuture<>();
+  }
+  
+  private static GpuTimeZoneDB instance = new GpuTimeZoneDB();
+  static GpuTimeZoneDB getInstance() {
```

Comment:
Should be `public static`?

Reply:
No this is default visiblity for testing purposes only. Added a comment.